File: python/leetcode/067.py
class Solution:
    def addBinary(self, a, b):
        i, j, carry, res = len(a) - 1, len(b) - 1, 0, ""
        while i > -1 or j > -1:
            num1 = ord(a[i]) - ord("0") if i > -1 else 0
            num2 = ord(b[j]) - ord("0") if j > -1 else 0
            i -= 1
            j -= 1
            res = str((carry + num1 + num2) % 2) + res
            carry = (carry + num1 + num2) / 2
        if carry > 0:
            res = str(carry) + res
        return res

File: cpp/leetcode/067.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int m = a.size() - 1, n = b.size() - 1;
        int carry = 0;
        string res = "";
        while (m >= 0 || n >= 0 || carry) {
            int x = (m >= 0) ? a[m--] - '0' : 0;
            int y = (n >= 0) ? b[n--] - '0' : 0;
            res = char((x + y + carry) % 2 + '0') + res;
            carry = (x + y + carry) / 2;
        }
        return res;
    }
};